extensions/ql-vscode/src/view/model-editor/ModelKindDropdown.tsx

```diff
@@ -66,19 +67,26 @@ export const ModelKindDropdown = ({
     [onChangeKind],
   );
 
+  const value =
+    modeledMethod && modeledMethodSupportsKind(modeledMethod)
+      ? modeledMethod.kind
+      : undefined;
+
   useEffect(() => {
-    const value = modeledMethod?.kind ?? "";
+    if (!modeledMethod || !modeledMethodSupportsKind(modeledMethod)) {
+      return;
+    }
 
     if (kinds.length === 0 && value !== "") {
       onChangeKind("");
-    } else if (kinds.length > 0 && !kinds.includes(value)) {
+    } else if (kinds.length > 0 && !kinds.includes(value ?? "")) {
       onChangeKind(kinds[0]);
     }
-  }, [modeledMethod?.kind, kinds, onChangeKind]);
+  }, [modeledMethod, value, kinds, onChangeKind]);
 
   return (
     <Dropdown
-      value={modeledMethod?.kind}
+      value={value ?? undefined}
```

Comment:
If `value` had the type `string | false` then this would coerce that to `string | undefined`. However, I believe the type of `value` is already `string | undefined`, so is `?? undefined` unneccesary here?

Reply:
Yep, it's unnecessary here, thanks for spotting that. I believe I first had `string | null` here, which was not an acceptable type for the `Dropdown` `value`.

I don't think `string | false` would actually be affected by this operator. `??` is different from `||` and will only change the result when the input is either `null` or `undefined`, not any falsy value. So, this was there to catch the `null` that was here before.